src/utils/notification-utils.spec.js

```diff
@@ -0,0 +1,178 @@
+import { beforeEach, describe, expect, it } from 'vitest';
+import { EVENT_TYPE, FAVORITE_TYPE, NOTIFICATION_REASON } from '../domain/constants.js';
+import { determineNotificationReasonsForRevisionCreatedEvent, determineNotificationReasonsForCommentCreatedEvent } from './notification-utils.js';
+
+describe('notification-utils', () => {
+
+  describe('determineNotificationReasonsForRevisionCreatedEvent', () => {
+    const testCases = [
+      {
+        description: 'when the notified user is the owner of the room',
+        event: { createdOn: new Date('2023-01-01'), type: EVENT_TYPE.revisionCreated, params: { userId: 'event-user-id' } },
+        revision: { _id: 'revision-id', documentId: 'document-id', roomId: 'room-id', roomContext: { draft: false } },
+        document: { _id: 'document-id', roomId: 'room-id', roomContext: { draft: false } },
+        room: { _id: 'room-id', owner: 'owner-user-id', members: [{ userId: 'notified-user-id' }] },
+        notifiedUser: { _id: 'owner-user-id', createdOn: new Date('2022-12-31'), favorites: [] },
+        expectedReasons: [NOTIFICATION_REASON.roomMembership]
+      },
```

Comment:
The negative cases of "notified user is the same as event user", "room/document/revision deleted" are not covered from what I can see, would be nice to have those too if it's not a hassle to add them.

Reply:
For the first scenario we do have both positive and negative cases:

* `when the notified user has marked the document as favorite`
* `when the user has marked the document as favorite but the notified user is the same user as the event user` (I forgot one `notified there`)

We do not have a test for deleted documents though, I'll add them (and correct the wording above).

Reply:
Added!

Reply:
👍